torch/_inductor/ir.py

```diff
@@ -3240,6 +3260,13 @@ def create(cls, kernel, *args, **kwargs):
             aten._linalg_svd.default,
             aten._linalg_svd.U,
             aten._fused_moving_avg_obs_fq_helper_functional,
+            # Here is the schema for this op:
+            #   _adaptive_avg_pool2d_backward(Tensor grad_output, Tensor self) -> Tensor
+            # if grad_out is contiguous while self is channels last,
+            # eager returns a channels last tensor while in FakeTensorMode we returns
+            # a contiguous tensor.
+            # This causes vgg16 training to fail when enabling layout optimization.
+            aten._adaptive_avg_pool2d_backward,
```

Comment:
@ezyang @eellison is there an easy way to fix the FakeTensor strides here?

Reply:
Yea, update [meta_registation](https://github.com/pytorch/pytorch/blob/main/torch/_meta_registrations.py#L1463) to reflect [cpu](https://github.com/pytorch/pytorch/blob/149237415ff12711cb4eca65c750a76f49872389/aten/src/ATen/native/AdaptiveAveragePooling.cpp#L82) and [cuda](https://github.com/pytorch/pytorch/blob/149237415ff12711cb4eca65c750a76f49872389/aten/src/ATen/native/cuda/AdaptiveAveragePooling.cu#L614) striding logic. Use [device hint](https://github.com/pytorch/pytorch/blob/main/torch/_meta_registrations.py#L1138) for separate cpu/cuda logic

Reply:
@eellison  I'm curious why we need different handling for different devices? Ed also mentioned the device hint thing above.

Reply:
We might not but sometimes the striding logic is different between cpu and cuda. I havent really checked yet for this op.